File: app/src/main/java/com/mc0239/recyclertableviewexample/database/AppDatabase.java
package com.mc0239.recyclertableviewexample.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import java.util.ArrayList;


@Database(entities = {User.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {

    public abstract UserDao userDao();

    private static volatile AppDatabase instance;

    public static AppDatabase getDatabase(final Context context) {
        if (instance == null) {
            synchronized (AppDatabase.class) {
                if (instance == null) {
                    instance = Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, "sample-db")
                            .allowMainThreadQueries()
                            .fallbackToDestructiveMigration()
                            .build();
                }
            }
        }
        return instance;
    }

    public void generateSampleData() {
        userDao().truncate();

        // Prepare some data for the table
        ArrayList<User> users = new ArrayList<>();
        for(int i = 0; i < 30; i++) {
            User u = new User();

            u.id = i;
            u.username = "johnd";
            u.name = "John";
            u.surname = "Doe ";

            users.add(u);
        }
        userDao().insert(users);
    }
}

File: app/src/main/kotlin/com/mc0239/recyclertableviewexample/database/AppDatabase.kt
package com.mc0239.recyclertableviewexample.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [User::class], version = 3)
abstract class AppDatabase : RoomDatabase() {

    abstract fun userDao(): UserDao

    companion object {
        private var instance: AppDatabase? = null

        @JvmStatic
        fun getDatabase(context: Context): AppDatabase {
            if (instance == null) {
                synchronized(AppDatabase::class.java) {
                    if (instance == null) {
                        instance = Room.databaseBuilder(context.applicationContext, AppDatabase::class.java, "sample-db")
                            .allowMainThreadQueries()
                            .fallbackToDestructiveMigration()
                            .build()
                    }
                }
            }

            return instance!!
        }
    }

    fun generateSampleData() {
        userDao().truncate()

        val users: MutableList<User> = mutableListOf()
        for (i in 0 until 30) {
            val u = User(i, "johnd", "John", "Doe", null)
            users.add(u)
        }
        userDao().insert(users)
    }

}